ttnn/cpp/ttnn/operations/uniform/device/kernels/writer.cpp

```diff
@@ -0,0 +1,73 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <cstdint>
+
+#include "common/constants.hpp"
+#include "dataflow_api.h"
+#include "debug/dprint.h"
+
+using namespace tt;
+
+void kernel_main() {
+    uint32_t dst_addr = get_arg_val<uint32_t>(0);
+    union {
+        float f;
+        uint32_t u;
+    } f2u_from, f2u_to;
+    f2u_from.u = get_arg_val<uint32_t>(1);
+    f2u_to.u = get_arg_val<uint32_t>(2);
+    uint32_t start_id = get_arg_val<uint32_t>(3);
+    uint32_t num_tiles = get_arg_val<uint32_t>(4);
+    uint32_t end_id = start_id + num_tiles;
+
+    constexpr uint32_t cb_intermed0_id = CB::c_intermed0;
+    constexpr uint32_t cb_out0_id = CB::c_out0;
+    const InterleavedAddrGenFast<true> output_addrg = {
+        .bank_base_address = dst_addr,
+        .page_size = get_tile_size(cb_out0_id),
+        .data_format = get_dataformat(cb_out0_id)};
+
+    uint32_t max_uint = 4294967295;
+    float random_range = f2u_to.f - f2u_from.f;
+
+    for (uint32_t i = start_id; i < end_id; ++i) {
+        cb_wait_front(cb_intermed0_id, 1);
+
+        uint32_t cb_intermed0_read_ptr = get_read_ptr(cb_intermed0_id);
+        uint32_t cb_out0_write_ptr = get_write_ptr(cb_out0_id);
```

Comment:
```C++
 * Returns a pointer to the beginning of a memory block previously reserved
 * by cb_reserve_back. Note that this call is only valid between calls
 * to cb_reserve_back and cb_push_back. The amount of valid memory
 * is equal to the number of tiles requested in a prior cb_reserve_back call.
 uint32_t get_write_ptr(uint32_t operand) {
```
https://github.com/tenstorrent/tt-metal/blob/main/tt_metal/hw/inc/dataflow_api.h#L399

According to the comment, `get_write_ptr` is only valid between `cb_reserve_back` and `cb_push_back`.
Even if there's no actual problem, it would be better to place cb_reserve_back and cb_push_back somewhere in the code for clarity and safety.

Reply:
I've added `cb_reserve_back`, but `cb_push_back` is not added as this cb is only used in 1 core for write purpose. If I add `cb_push_back` the program will hang as there is no other core executing `cb_wait_front`. The same approach can be founded in this file: ttnn/cpp/ttnn/operations/data_movement/fill_rm/device/kernels/dataflow/fill_rm_interleaved.cpp

Reply:
I understand that we only need to add `cb_reserve_back`. However, I don't see `cb_reserve_back` in the code. It seems that the commit might be missing it.

Reply:
`cb_reserve_back` is added in this [commit](https://github.com/tenstorrent/tt-metal/pull/13735/commits/f07514be68e8d056e929063eebf882750f50e812#diff-094c033c84cbdc24390a2b6b167f2531cd87ffb3309689586f37916fc0fdbdb0), line 37

Reply:
Got it. It seems I might have been looking at an outdated file.